src/ecs/Engine.hx

```diff
@@ -337,4 +341,41 @@ class Engine {
 
         return target_system;
     }
+
+    public function addGlobalSystem<T:ecs.System>(systemClass:Class<T>) {
+        var target_system:System = getOrCreateSystem(systemClass);
+        globalSystems.push(target_system);
+        enable(systemClass);
+    }
+
+    public function setStateSystem<T:ecs.System>(state:Int, systemClass:Class<T>) {
```

Comment:
Any reason to use Int for state instead of string?

Reply:
The idea is to be able to use an enum in the user code, g.e :
```haxe
enum abstract AppState(Int) from Int to Int {
    var Intro = 0;
    var Menu;
    var InGame;
}
```
I'm open to any solution but I need something statically checked

Reply:
In fact I can do the same with string, code updated